worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/AddWorkflowConfigs.java

```diff
@@ -0,0 +1,289 @@
+import com.github.cafdataprocessing.processing.service.client.model.EffectiveTenantConfigValue;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsCacheKey;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsRepositoryCacheKey;
+import com.github.cafdataprocessing.workflow.cache.WorkflowSettingsTenantCacheKey;
+import static com.github.cafdataprocessing.workflow.transform.TransformerFunctions.LOG;
+import static com.github.cafdataprocessing.workflow.transform.TransformerFunctions.escapeForJavaScript;
+import com.google.common.base.Strings;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.gson.Gson;
+import java.lang.reflect.Type;
+import com.google.gson.reflect.TypeToken;
+import com.hpe.caf.worker.document.exceptions.DocumentWorkerTransientException;
+import com.hpe.caf.worker.document.model.Document;
+import com.sun.jersey.api.client.ClientHandlerException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public final class AddWorkflowConfigs
+{
+    private final static Logger LOGGER;
+    private final static ApiClient API_CLIENT;
+    private final static LoadingCache<WorkflowSettingsCacheKey, String> SETTINGS_CACHE;
+    private final static Gson GSON;
+    private final static RepositoryConfigurationApi REPOSITORY_CONFIG_API;
+    private final static TenantConfigurationApi TENANT_CONFIG_API;
+
+    static {
+        LOGGER = LoggerFactory.getLogger(AddWorkflowConfigs.class);
+        API_CLIENT = new ApiClient();
+        SETTINGS_CACHE = CacheBuilder.newBuilder()
+            .maximumSize(100)
+            .expireAfterAccess(5, TimeUnit.MINUTES)
+            .recordStats()
+            .build(
+                new CacheLoader<WorkflowSettingsCacheKey, String>()
+            {
+                @Override
+                public String load(final WorkflowSettingsCacheKey key)
+                    throws ApiException, DocumentWorkerTransientException
+                {
+                    LOGGER.debug("Key not found in cache: " + key);
+                    return getSettingsFromServer(key);
+                }
+            });
+        GSON = new Gson();
+        final String processingSericeBaseUrl = System.getenv("CAF_WORKFLOW_WORKER_PROCESSING_API_URL");
+        Objects.requireNonNull(processingSericeBaseUrl);
+        API_CLIENT.setBasePath(processingSericeBaseUrl);
+        TENANT_CONFIG_API = new TenantConfigurationApi(API_CLIENT);
+        REPOSITORY_CONFIG_API = new RepositoryConfigurationApi(API_CLIENT);
+    }
+
+    public static void addCustomWorkflowConfig(final String requiredConfig, final Document document)
```

Comment:
Convert the JSON to an object in the WorkflowTransformer. Introduce some classes to make the settings more strongly typed. A Map<String, Map<String, String>> is hard to use. Create a workflow settings object with properties for task, repository and tenant. Change the lists of objects to use a more suitable type. String for tenant settings or create a class if the setting is more complex.

This code would be better in WorkflowTransformer.java, 

```
        final Type type = new TypeToken<Map<String, List<Object>>>(){}.getType();
        final Map<String, List<Object>> requiredConfigMap = GSON.fromJson(requiredConfig, type);
        final List<Object> taskConfigs = requiredConfigMap.get("taskSettings");
        final List<Object> repositoryConfigs = requiredConfigMap.get("repositorySettings");
        final List<Object> tenantConfigs = requiredConfigMap.get("tenantSettings");
        final Map<String, Map<String, String>> settings = new HashMap<>();
        settings.put("task", processTaskConfigs(document, taskConfigs));
        settings.put("repository", processRepositoryConfigs(document, tenantId, repositoryConfigs));
        settings.put("tenant", processTenantConfigs(tenantId, tenantConfigs));
```